🚀 2단계 - 로그인 리팩터링(김준성)

junseong94 opened the pull request:
안녕하세요 리뷰어님 :)

step2 로그인 리팩터링 완료하여 PR 드립니다.

오늘도 좋은하루 되시길 바랍니다 ㅎㅎ

junseong94 committed:
rebase(project) : 프로젝트 리베이스

PR된 원격저장소 기준 rebase 수행

junseong94 committed:
rebase : 프로젝트 rebase

PR된 원격저장소 기준 rebase 수행

junseong94 committed:
rebase : 프로젝트 rebase

PR된 원격저장소 기준 rebase 수행

junseong94 committed:
chore(class) : 불필요한 클래스 파일 삭제

- rebase 과정에서 불필요한 남아있는 클래스 파일 삭제

junseong94 committed:
refactor(ResponseEntity) : ResponseEntity.create() 적용

- 메서드 통일을 위해 create()로 리팩토링

junseong94 committed:
chore(convention) : 패키지 명 변경

- 구글 Java Style Guide에 기준(lowercase)으로 패키지명 변경
  - GlobalFixture to globalfixture
  - reservationTheme to reservationtheme
  - reservationTime to reservationtime

junseong94 committed:
refactor(test) : 테스트 코드 리팩토링

- MissionStepTest 이름 변경 (to ReservationIntegrationTest)
- root('/') 호출 테스트 코드 RedirectPageTest로 이동
- ReservationIntegrationTest 에 예약 생성, 삭제 조건으로 테스트 코드 재구성

junseong94 committed:
refactor(test) : 테스트 코드 리팩토링

- @ParameterizedTest 적용으로 중복 코드 제거

junseong94 committed:
refactor(test) : 테스트 코드 리팩토링

- assertThat 코드가 2개이상일 경우 assertAll과 assertSoftly 적용

junseong94 committed:
refactor(test) : MemberIntegrationTest 리팩토링

- 불필요한 코드줄 제거
- @ParamaterizedTest 적용
  - basiceLoginException()
- 클래스명 변경 (MemberControllerTest to MemberIntegrationTest)

junseong94 committed:
refactor(test) : ReservationTimeControllerTest 리팩토링

- 두 개 이상의 assertThat() 코드는 assertSoftly() 적용
- 시간 삭제 테스트에서 시간 생성 로직은 중요도가 낮아 creatTime() 메서드로 extract

junseong94 committed:
fix(final keyword) : 누락된 final 코드줄 키워드 적용

- repository 코드중 파라미터에 final 누락되어 적용

junseong94 committed:
refactor : ui 활성화를 코드 변경

- step2 진행을 위해 화면 ui 코드 변경

junseong94 committed:
fix : ui 활성화를 코드 변경

- step2 진행을 위해 화면 ui 코드 변경

junseong94 committed:
test(schema): 예약시간, 예약테마 테스트 데이터 입력

- ui 테스트 수행을 위해 초기 데이터 입력

junseong94 committed:
style(fields name): 필드 명 변경

- 클라이언트와 변수 매핑을 동일하게 하기위해 변수명과 함수명 변경
  - id to timeId, getId() to getTimeId()
  - id to themeId, getId() to getThemeId()

junseong94 committed:
style(fields name): 필드 명 변경

- 클라이언트와 변수 매핑을 동일하게 하기위해 변수명과 함수명 변경
  - id to timeId, getId() to getTimeId()
  - id to themeId, getId() to getThemeId()

junseong94 committed:
style(fields name): 필드 명 변경

- 클라이언트와 변수 매핑을 동일하게 하기위해 변수명과 함수명 변경
  - id to timeId, getId() to getTimeId()
  - id to themeId, getId() to getThemeId()

junseong94 committed:
style: 빈 줄 제거와 코드 align 수정

- 불필요한 빈 줄 제거 및 가독성을 위한 코드 align 수정

junseong94 committed:
refactor: setter 및 toString 메서드 추가

- 요구사항 변경에 따른 setter() 메서드 추가
- 디버깅을 위한 toString() 메서드 추가

junseong94 committed:
refactor: 토큰 추출 메서드 공통메서드로 extract

- ArgumentResolver에서 토큰 추출 메서드 사용으로 인한 공통 메서드로 extract

junseong94 committed:
feat(ArgumentResolver): 요구사항 추가에 따른 ArgumentResolver 추가

- 요구사항: 예약등록 시 토큰정보를 기반으로 회원정보 입력
  - ArgumentResolver 구현 및 어노테이션 추가
  - 추가한 ArgumentResolver를 WebMvcConfigurer 으로 등록

junseong94 committed:
styel(람다식 변수명): 변수명 수정 (m to savedMember)

- 코드 통일성과 가독성을 위해 람다 변수명 수정

junseong94 committed:
feat(dto): MemberRequestDto 추가

- ArgumentResolver 추가에 따른 MemberRequestDto 추가

junseong94 committed:
style(class name): rename MemberController to LoginMemberController

- 추후 "/members" url 개발 대비 controller 명 변경

junseong94 committed:
chore: 사용하지 않는 함수 및 로그 줄 제거

- 사용하지 않는 함수 및 로그 줄 제거

junseong94 committed:
refactor: dto에 @JsonProperty 적용

- 클라리언트 변수와 동일한 매핑을 유지하기 위해 @JsonProperty 적용

pkeugine commented on src/main/java/roomescape/reservation/dto/ReservationResponseDto.java:
```diff
@@ -1,30 +1,37 @@
+import roomescape.reservationtheme.dto.ReservationThemeResponseDto;
+import roomescape.reservationtime.dto.ReservationTimeResponseDto;
 
 public class ReservationResponseDto {
```
해당 dto 로 인해 테스트가 깨지고 있는데, 한 번 확인해주시면 좋을 것 같아요.
리뷰 요청 전에 테스트를 실행해보고 빌드에 이상이 없는지 꼭 확인해주세요!

pkeugine commented on src/test/java/roomescape/reservation/ReservationIntegrationTest.java:
```diff
@@ -1,39 +1,34 @@
+                "브라운", "2025-08-05", reservationTimeRequestDto, reservationThemeRequestDto);
+        // 시간 등록
         RestAssured.given().log().all()
                 .contentType(ContentType.JSON)
```
before each 메서드에서는 검증을 수행하기보다는 다른 방식으로 데이터를 넣어주는건 어떨까요?

pkeugine commented on src/test/java/roomescape/reservation/ReservationIntegrationTest.java:
```diff
@@ -42,20 +37,19 @@ void reservation() {
                 .statusCode(200)
                 .body("id", is(1));
```
해당 부분에 문제가 있어서 관련 테스트가 다 통과되고 있지 않아요.
수정해보면 좋을 것 같네요!

pkeugine commented on src/test/java/roomescape/member/MemberIntegrationTest.java:
```diff
@@ -53,23 +54,8 @@ void basicLoginException1() {
-        assertThat(responseBody2).isEqualTo("Authorization Exception: 해당하는 회원 정보가 없습니다.");
     }
 
     @DisplayName("쿠키를 이용하여 로그인 사용자 이름을 응답받습니다.")
```
해당 테스트가 deserialization 문제 때문에 실패하고 있어요!

pkeugine commented on src/main/java/roomescape/member/dto/MemberResponseDto.java:
```diff
@@ -1,12 +1,11 @@
 package roomescape.member.dto;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
```
사용하지 않는 import 문은 제거하도록 하죠
다른 클래스에서도 종종 보이니 프로젝트 전반적으로 한 번 확인해주세요!

pkeugine commented on src/test/java/roomescape/reservation/ReservationControllerTest.java:
```diff
@@ -117,8 +125,12 @@ void createReservationEmptyName(final String name) {
+        assertSoftly(
+                softAssertions -> {
+                    assertThat(response.statusCode()).isEqualTo(HttpStatus.BAD_REQUEST.value());
+                    assertThat(response.jsonPath().getString("name")).isEqualTo("예약자 명 입력해주세요");
```
해당 테스트 역시 실패하고 있어요. 한 번 확인해주시기 바랍니다.

pkeugine commented on src/test/java/roomescape/reservation/ReservationControllerTest.java:
```diff
@@ -7,26 +7,27 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
 
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 @TestPropertySource(locations = "classpath:test-application.yml")
```
`ReservationControllerTest` 가 전반적으로 실패하는 테스트가 더 많은 것 같아요.
다행히 에러 로그를 보면 어느정도 문제가 되는 부분을 잘 알아볼 수 있어서 전반적으로 확인해주시면 좋겠네요.

pkeugine commented on src/main/java/roomescape/member/application/AuthService.java:
```diff
@@ -41,10 +41,10 @@ public MemberResponseDto findMemberName(final String token) {
-
     }
+
+
```
불필요한 공백이 포함됐어요!

pkeugine commented on src/main/java/roomescape/member/dto/MemberRequestDto.java:
```diff
@@ -0,0 +1,46 @@
+    private Long id;
+    private String name;
+    private String email;
+    private String password;
```
`MemberRequestDto` 에서는 name 만 사용하고 있네요!
그렇다면 나머지 불필요한 필드는 없애보면 어떨까요?
사용하지 않는다면 꼭 미리 추가해둘 필요는 없으니까요.

pkeugine commented on src/main/java/roomescape/member/dto/MemberRequestDto.java:
```diff
@@ -0,0 +1,46 @@
+        this.name = name;
+        this.email = email;
+        this.password = password;
+    }
```
마찬가지로 사용하지 않는 생성자도 제거해볼 수 있겠네요!

pkeugine commented on src/main/java/roomescape/member/infra/CommonMethod.java:
```diff
@@ -0,0 +1,15 @@
+
+import jakarta.servlet.http.Cookie;
+
+public class CommonMethod {
```
해당 클래스명을 조금 더 구체적으로 바꿔봐도 괜찮을 것 같아요.
토큰에서 쿠키를 추출하는 일을 하고 있으니, 관련된 유틸리티 클래스라는 것을 나타내보면 좋겠네요!

pkeugine commented on src/main/java/roomescape/member/infra/MemberArgumentResolver.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    @Override
+    public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer, NativeWebRequest webRequest, WebDataBinderFactory binderFactory) throws Exception {
```
```suggestion
    public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer, NativeWebRequest webRequest, WebDataBinderFactory binderFactory) {
```

`throws Exception` 부분은 불필요하므로 제거해보면 좋을 것 같아요.
그리고 굉장히 길어서 줄바꿈을 적용하면 읽기 더 편하겠네요!

pkeugine commented on src/main/java/roomescape/member/infra/MemberConfiguration.java:
```diff
@@ -0,0 +1,22 @@
+
+    private final MemberArgumentResolver memberArgumentResolver;
+
+    public MemberConfiguration(MemberArgumentResolver memberArgumentResolver) {
+        this.memberArgumentResolver = memberArgumentResolver;
+    }
```
final 이 누락된 것 같아요
종종 보이는 곳이 있으니, 컨벤션을 한 번 잘 지켜보도록 하죠!
사실 final 을 꼼꼼히 붙이는 컨벤션은 저도 잘 지키지 못하는 어려운 작업이에요 ㅋㅋㅋ

pkeugine commented on src/main/java/roomescape/member/infra/MemberRepository.java:
```diff
@@ -30,7 +30,7 @@ public Long findIdByEmail(final String email) {
     public String findNameById(final Long id) {
         return MemberStorage.values().stream()
-                .filter(m -> m.getId().equals(id))
+                .filter(savedMember -> savedMember.getId().equals(id))
```
```suggestion
                .filter(savedMember -> savedMember.hasId(id))
```
이런식으로 멤버가 직접 확인하게 해보는건 어떨까요?

pkeugine commented on src/main/java/roomescape/reservationtheme/domain/ReservationTheme.java:
```diff
@@ -1,22 +1,24 @@
 public class ReservationTheme {
 
-    private Long id;
+    private Long themeId;
```
이름에 도메인명을 붙이는 방식으로 수정해주신 이유가 궁금해요!
(틀렸다는건 아니고, 궁금증입니다)

junseong94 replied:
클라이언트 변수명과 맞출려고 바꿨었는데 
생각해보니 dto에서 @JsonProperty로만 잡아주면 굳이 변경할 필요가 없더라구요.

그래서 다시 원복했습니다 ㅎㅎ;;

pkeugine commented on src/main/java/roomescape/reservationtime/dto/ReservationTimeRequestDto.java:
```diff
@@ -0,0 +1,39 @@
+
+import jakarta.validation.constraints.NotBlank;
+
+public class ReservationTimeRequestDto {
+
+    private Long timeId;
+    @NotBlank(message = "예약 시간을 입력해주세요")
+    private String startAt;
+
+    public ReservationTimeRequestDto(Long timeId) {
+        this.timeId = timeId;
+    }
+
+    public ReservationTimeRequestDto(String startAt) {
+        this.startAt = startAt;
+    }
```
하나의 dto 를 reservation 과 reservation time 관련 로직에 재활용하느라
상황에 따라 사용하기도 하고, 사용하지 않는 필드가 함께 모여있네요.
지금은 필드가 두 개라 그리 복잡하진 않지만,
서비스가 커질수록 이는 NPE 를 발생시키는 문제라던지, 개발자가 실수할 가능성일 높일 것 같아요.
유지보수 할 때도 응집도가 떨어질 수도 있구요.
`ReservationTimeRequestDto` 는 말 그래도 ReservationTime 을 생성할 때만 사용하도록 만드는건 어떨까요?

pkeugine commented on src/main/java/roomescape/member/infra/MemberArgumentResolver.java:
```diff
@@ -0,0 +1,42 @@
+@Component
+public class MemberArgumentResolver implements HandlerMethodArgumentResolver {
+
+    private static final Logger log = LoggerFactory.getLogger(MemberArgumentResolver.class);
+    private final AuthService authService;
```
static 이 붙은것과 그렇지 않은 것 사이에 공백을 두면 읽을 때 실수 없이 구분하여 읽기 편할 것 같아요 :)

pkeugine commented on src/main/java/roomescape/member/infra/MemberRepository.java:
```diff
@@ -30,7 +30,7 @@ public Long findIdByEmail(final String email) {
 
```
`isExistByEmail` 메서드는 사용하고 있지 않으니 제거합시다!
제거하자는 리뷰를 꽤 많이 남기는 것 같은데,
생각보다 중요한 부분이라고 생각해요.
집 청소를 안하다보면 한 없이... 쓰레기와 청소할 것들이 쌓여가듯이, 프로젝트도 레거시로 남아있게 되는 것 같거든요.

pkeugine commented on src/main/java/roomescape/reservation/application/ReservationService.java:
```diff
@@ -1,15 +1,16 @@
 package roomescape.reservation.application;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
```
전체적으로 로깅을 적극적으로 활용하시는 것 같아요! 👍 
시도했다가 빼게 된다면 잊지 않고 import 문을 제거해주죠!

pkeugine commented on src/main/java/roomescape/reservation/application/ReservationService.java:
```diff
@@ -73,7 +69,7 @@ public List<ReservationTimeResponseDto> findAvaliableTimes(final String date, fi
         final List<ReservationTime> availableReservationTimes = reservationRepository.getAvailableReservationTimes(date, themeId);
```
`findById` 도 제거해줍시다

pkeugine replied:
`findAvaliableTimes` -> `findAvailableTimes`

pkeugine commented on src/main/java/roomescape/reservation/domain/Reservation.java:
```diff
@@ -1,7 +1,7 @@
+import roomescape.reservationtime.domain.ReservationTime;
 
 import java.util.Objects;
 
```
못 보고 넘어간 부분이 있는데, `reservationPolicy` 는 static 하니 `RESERVATION_POLICY` 라고 이름을 지어도 괜찮을 것 같아요.

pkeugine replied:
(+) 테스트에서만 사용되는 생성자는 지양해보는건 어떨까요?

pkeugine commented on src/main/java/roomescape/member/infra/MemberRepository.java:
```diff
@@ -30,7 +30,7 @@ public Long findIdByEmail(final String email) {
+                .filter(savedMember -> savedMember.getId().equals(id))
                 .map(Member::getName).findFirst().orElse(null);
     }
 }
```
`MemberRepository` 도 이제 db 를 사용하도록 바꿔보면 좋을 것 같네요!

pkeugine commented on src/main/java/roomescape/reservation/dto/ReservationRequestDto.java:
```diff
@@ -68,4 +65,18 @@ public ReservationTimeRequestDto getReservationTimeRequestDto() {
         return reservationThemeRequestDto;
     }
+
+    public void setName(@NotBlank(message = "예약자명을 입력해주세요") String name) {
+        this.name = name;
+    }
```
`set` 이라는 이름 대신 다른 표현을 사용해보면 어떨까요?
결국 setter 와 하는 행위가 같긴하지만,
이름을 다르게 하는 것만으로도 해당 메서드가 조금 더 범용적으로 사용될 수 있게 한다거나
만드는 사람 / 사용하는 사람에게 경각심을 줄 수 있거든요.

pkeugine reviewed (changes requested):
안녕하세요 준성님,
꽤나 많은 작업이 포함되어 있네요! 😮 
고생 많으셨습니다.

리뷰 한 번 읽어보시고, 답변을 추가하거나 반영해보시면 좋을 것 같아요.
테스트나 코드 일관성, 유지보수 관련된 리뷰가 대부분이라 쉽게 작업하실 수 있을거예요.

혹시라도 이해가 되지 않거나 다른 의견이 있다면 언제는 코멘트 환영입니다!
곧 주말이 다가오니 조금만 더 힘내보도록 하죠 😄 
파이팅입니다!!! 🔥 🔥 🔥

junseong94 committed:
chore(schema): member 테이블 스키마 추가

- 인메모리 저장소에서 데이터베이스로 변경으로 인한 스키마 추가

junseong94 committed:
refactor(Repository): Member 엔티티의 인메모리 저장소를 데이터베이스로 변경, 그에따른 코드 변경

- 스트림 to jdbcTemplate

junseong94 committed:
chore: convention 에 따른 코드 정리

- 불필요한 코드 줄 제거
- typo. 수정(findAvaliableTimes to findAvailableTimes)
- static 변수명은 대문자로 변경
- 가독성을 위한 긴 줄 코드 정리
- 파라미터 final 추가
- 사용하지 않는 import 제거
- 사용하지 않는 생성자, 메서드 제거
- themeId, timeId 필드명 id로 원복

junseong94 committed:
refactor(Member): 데이터베이스 적용에 따른 엔티티 변경

- ATOMIC_LONG 제거

junseong94 committed:
chore: MemberRequestDto 에 사용하지 않는 필드, 생성자 제거

- 필드 제거: id, email, password
- 생성자, toString() 제거
- getter 제거 : email, password

junseong94 committed:
chore: ReservationRequestDto Name Setter 이름 변경

- setName to assignName

junseong94 committed:
chore: CommonMethod 클래스명 변경

- CommonMethod to TokenUtil

junseong94 committed:
test(schema): Member 의 데이터베이스 적용에 따른 테스트 schema 변경

- member table 및 초기 데이터 추가

junseong94 committed:
test(member): LoginMemberDtoFixture 생성

- 토큰 반환용 static 메서드 getLoginMemberToken() 생성 및 연관 테스트 코드 변경
- createLoginMember() 생성 및 연관 테스트 코드 변경

junseong94 committed:
refactor: ReservationTimeRequestDto 역할분리

- ReservationTime 전환을 위한 역할만 하도록 연관 코드 수정
- Reservation 에 연관된 코드는 TimeDto 를 생성해 적용
- 관련 테스트 코드 수정

junseong94 committed:
chore: 필드명 원복 및 @JsonProperty 적용

- themeId, timeId to id 로 원복
- ReservationThemeRequestDto 에 @JsonProperty 적용

junseong94 committed:
test : LoginMemberDtoFixture 적용

- createLoginMember()

pkeugine reviewed (approved):
안녕하세요 준성님!
리팩토링 잘 진행해주셨고,
테스트도 다 통과되어 다음 스텝으로 넘어가서 얘기 나눠보면 좋을 것 같아요.
힘들었을 수도 있는데 깔끔하게 바꿔주셨네요 👍 
남은 금요일과 주말 잘 보내시고, 조만간 다시 뵙도록 하겠습니다!
조금만 더 파이팅해요! 🔥 🔥 🔥
